### PR TITLE
Fixing #149: Send `Origin` header for outgoing requests (AWS EC2 needs this)

### PR DESCRIPTION
**Pull request by mozfreddyb**
@fmarier Can you take a look? I'm quite uncertain about how to test this.

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -128,7 +128,10 @@ var fetchResource = function (resourceUrl, cb) {
     return cb(false);
   }
 
-  moduleForProtocol(resourceUrl).get(resourceUrl, function (response) {
+  moduleForProtocol(resourceUrl).get({
+    hostname: resourceUrl,
+    headers: { 'Origin': 'https://srihash.org/' },
```
Comment (fmarier) [inline lib/helpers.js]: Maybe instead of hard-coding the URL we should use hapi's `server.info.url`?

**Comment by fmarier** [comment]
```
  1) generate() "before all" hook:

     Uncaught Error: getaddrinfo ENOTFOUND https://code.jquery.com/jquery-1.11.2.min.js https://code.jquery.com/jquery-1.11.2.min.js:443

      at errnoException (dns.js:27:10)

      at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:78:26)

  2) generateElement() "before all" hook:

     Uncaught Error: getaddrinfo ENOTFOUND https://www.google-analytics.com/ga.js https://www.google-analytics.com/ga.js:443

      at errnoException (dns.js:27:10)

      at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:78:26)
```

It looks like our Travis container no longer has access to the network?

File: lib/helpers.js (hunk 1 above)
Comment (mozfreddyb) [inline lib/helpers.js]: Yep, that would be great.
But I couldn't find a place in the helpers code that seems like a good point to get to `server` (as defined in index.js).

Comment (fmarier) [inline lib/helpers.js]: I guess we should file a follow-up bug if we end up merging this with the hard-coded string.

**Comment by XhmikosR** [comment]
@mozfreddyb: have you tried restarting the build?

Note that I made a similar change in Travis in #162 without having seen this, so if that gets merged you can rebase this one.

**Comment by XhmikosR** [comment]
@mozfreddyb: I believe your code is wrong now that i looked closer.

From https://nodejs.org/api/http.html#http_http_get_options_callback:

>options \<Object\> | \<string\> | \<URL\> Accepts the same options as http.request(), with the method always set to GET. Properties that are inherited from the prototype are ignored.

So, here you are passing an Object, with the full URL as the `hostname`.

**Comment by XhmikosR** [comment]
So, basically, this should work:

```diff
 lib/helpers.js | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/lib/helpers.js b/lib/helpers.js
index 441473a..466ddb0 100644
--- a/lib/helpers.js
+++ b/lib/helpers.js
@@ -128,7 +128,15 @@ var fetchResource = function (resourceUrl, cb) {
     return cb(false);
   }
 
-  moduleForProtocol(resourceUrl).get(resourceUrl, function (response) {
+  var urlObject = url.parse(resourceUrl);
+
+  var options = {
+    hostname: urlObject.hostname,
+    path: urlObject.path,
+    headers: { 'Origin': 'https://srihash.org/' }
+  };
+
+  moduleForProtocol(resourceUrl).get(options, function (response) {
     var output;
     if (response.headers['content-encoding'] === 'gzip') {
       var gzip = zlib.createGunzip();

```

**Comment by XhmikosR** [comment]
Oh, and the Travis change shouldn't be needed :)

**Comment by XhmikosR** [comment]
I pushed my patch in my #164 PR, because I'm too lazy to split plus I want us to merge the other stuff :) Only thing missing is a proper test.

**Comment by fmarier** [comment]
Superseded by #177.

**Pull request closed without merging**